File: flipt-java/src/main/java/com/flipt/api/authentication/ClientTokenAuthenticationStrategy.java
package com.flipt.api.authentication;

public final class ClientTokenAuthenticationStrategy implements AuthenticationStrategy {
  private final String clientToken;

  public ClientTokenAuthenticationStrategy(String clientToken) {
    this.clientToken = clientToken;
  }

  @Override
  public String getAuthorizationHeader() {
    return String.format("Bearer %s", this.clientToken);
  }
}

File: flipt-csharp/src/Authentication/ClientTokenAuthenticationStrategy.cs
namespace Flipt.Authentication;

/// <summary>
/// This class is responsible for providing the client token authentication strategy.
/// </summary>
public class ClientTokenAuthenticationStrategy : IAuthenticationStrategy
{
    private readonly string _clientToken;

    public ClientTokenAuthenticationStrategy(string clientToken)
    {
        _clientToken = clientToken;
    }
    public KeyValuePair<string, string> GetAuthorizationHeader()
    {
        return new KeyValuePair<string, string>("Bearer", _clientToken);
    }
}